fastmri_recon/models/utils/fourier.py

```diff
@@ -182,16 +184,26 @@ def op(self, inputs):
 
     def adj_op(self, inputs):
         if self.multicoil:
-            raise NotImplementedError('Multicoil NFFT is not implemented yet.')
-        elif self.density_compensation:
-            kspace, ktraj, shape, dcomp = inputs
+            if self.density_compensation:
+                kspace, ktraj, smaps, shape, dcomp, = inputs
+            else:
+                kspace, ktraj, smaps, shape = inputs
         else:
-            kspace, ktraj, shape = inputs
+            if self.density_compensation:
+                kspace, ktraj, shape, dcomp = inputs
+            else:
+                kspace, ktraj, shape = inputs
         shape = tf.reshape(shape[0], [])
-        if self.density_compensation:
-            kspace = tf.cast(dcomp, kspace.dtype) * kspace[..., 0]
+        if self.multicoil:
+            if self.density_compensation:
+                kspace = tf.cast(dcomp, kspace.dtype) * kspace[..., 0]
+            else:
+                kspace = kspace[..., 0]
         else:
-            kspace = kspace[..., 0]
+            if self.density_compensation:
+                kspace = tf.cast(dcomp, kspace.dtype) * kspace
```

Comment:
Why don't you have `kspace[..., 0]` here?

Reply:
Oops, I think I messed the code here, you WILL need it here, you will not need it for multichannel case (above)

Reply:
I am surprised : why don't you need it in the multichannel case?

Reply:
I think you are right, I misunderstood your concern, while I need to run codes to check this, I think we need it in both the cases.

Reply:
This can be resolved right?

Reply:
yes